cmd/source_profile.go

```diff
@@ -0,0 +1,283 @@
+			conn.ty = SourceProfileConnectionTypePostgreSQL
+			conn.pg = NewSourceProfileConnectionPostgreSQL(params)
+		}
+	case "dynamodb":
+		{
+			conn.ty = SourceProfileConnectionTypeDynamoDB
+			dydb, err := NewSourceProfileConnectionDynamoDB(params)
+			conn.dydb = dydb
+			if err != nil {
+				return conn, err
+			}
+		}
+	default:
+		return conn, fmt.Errorf("please specify a valid source database using -source flag, received source = %v", source)
+	}
+	return conn, nil
+}
+
+type SourceProfileConfig struct {
+	path string
+}
+
+func NewSourceProfileConfig(path string) SourceProfileConfig {
+	return SourceProfileConfig{path: path}
+}
+
+type SourceProfile struct {
+	ty     SourceProfileType
+	file   SourceProfileFile
+	conn   SourceProfileConnection
+	config SourceProfileConfig
+}
+
+func (src SourceProfile) ToLegacyDriver(source string) (string, error) {
+	switch src.ty {
+	case SourceProfileTypeFile:
+		{
+			switch strings.ToLower(source) {
+			case "mysql":
+				return "mysqldump", nil
+			case "postgresql", "postgres", "pg":
+				return "pg_dump", nil
+			case "dynamodb":
+				return "", fmt.Errorf("dump files are not supported with DynamoDB")
+			default:
+				return "", fmt.Errorf("please specify a valid source database using -source flag, received source = %v", source)
+			}
+		}
+	case SourceProfileTypeConnection:
+		{
+			switch strings.ToLower(source) {
+			case "mysql":
+				return "mysql", nil
+			case "postgresql", "postgres", "pg":
+				return "postgres", nil
+			case "dynamodb":
+				return "dynamodb", nil
+			default:
+				return "", fmt.Errorf("please specify a valid source database using -source flag, received source = %v", source)
+			}
+		}
+	case SourceProfileTypeConfig:
+		return "", fmt.Errorf("specifying source-profile using config not implemented")
+	default:
+		return "", fmt.Errorf("invalid source-profile, could not infer type")
+	}
+}
+
+// Source profile is passed as a list of key value pairs on the command line.
+// Following 3 formats are supported as valid source profiles.
+//
+// Format 1. Specify file path and file format.
+// File path can be a local file path or a gcs file path. Support for more file
+// path types can be added in future.
+// File format can be "dump" e.g., when specifying a mysqldump or pgdump etc.
+// Support for more formats e.g., "csv", "avro" etc can be added in future.
+//
+// Example: -source-profile="file=/tmp/abc, format=dump"
+// Example: -source-profile="file=gcs://bucket_name/cart.txt, format=dump"
+//
+// Format 2. Specify source connection parameters. If none specified, then read
+// from envrironment variables.
+//
+// Format 3. Specify a config file that specifies source connection profile.
+//
+func NewSourceProfile(s string, source string) (SourceProfile, error) {
+	if source == "" {
+		return SourceProfile{}, fmt.Errorf("cannot leave -source flag empty, please specify source databases e.g., -source=postgres etc")
+	}
+
+	params := make(map[string]string)
+
+	if len(s) > 0 {
```

Comment:
we should use some existing library to parse csv instead, to handle scenarios where the parameters have commas within the quotes

Reply:
Done.